### PR TITLE
Allow all trackers committing into DB

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

Historically only `account updates` tracker was committing into DB. The PR changes this:
1. `dbRound` management and `commitSyncer` business moved to trackerRegistry.
2.  `account updates` is used as a driver for committing tasks scheduling - by given `commttedRound` and `dbRound` it figures out offset in detlas for saving to the disk. This is basically old `au.committedUpTo` method`.
3. `commitRound` replaced by `prepareCommit` that returns commit and post commit callbacks
4. trackerRegistry has own `commitRound` that calls  trackers' `prepareCommit` and commit/post commit procs if any.
5. `account updates` still have a cached dbRound to be used in Lookup methods.

## Test Plan

It is refactoring, use existing tests

### CODE REVIEW

File: ledger/bulletin.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func (b *bulletin) Wait(round basics.Round) chan struct{} {
 	return signal.signal
 }
 
-func (b *bulletin) loadFromDisk(l ledgerForTracker) error {
+func (b *bulletin) loadFromDisk(l ledgerForTracker, dbRound basics.Round) error {
```
Comment (tsachiherman) [inline ledger/bulletin.go:81]: dbRound -> _

File: ledger/tracker.go [hunk 2]
```diff
@@ -98,19 +116,73 @@ type ledgerForTracker interface {
...
+	scheduleCommit(basics.Round)
+	waitAccountsWriting()
 }
 
 type trackerRegistry struct {
 	trackers []ledgerTracker
+	driver   *accountUpdates
+
+	// ctx is the context for the committing go-routine. It's also used as the "parent" of the catchpoint generation operation.
+	ctx context.Context
```
Comment (tsachiherman) [inline ledger/tracker.go:130]: I don't see that this context object is the parent of the catchpoint generation operation; did I miss it ?

File: ledger/tracker.go [hunk 3]
```diff
@@ -98,19 +116,73 @@ type ledgerForTracker interface {
...
+
+	tr.ctx, tr.ctxCancel = context.WithCancel(context.Background())
+	tr.committedOffset = make(chan deferredCommit, 1)
+	tr.commitSyncerClosed = make(chan struct{})
+	go tr.commitSyncer(tr.committedOffset)
+
+	return
 }
 
 func (tr *trackerRegistry) register(lt ledgerTracker) {
```
Comment (tsachiherman) [inline ledger/tracker.go:175]: the `register` can be combined with the `initialize` method. In particular, you can just pass in a slice of `ledgerTracker` object pointers.

File: ledger/tracker.go (hunk 2 above)
Comment (algorandskiy) [reply ledger/tracker.go:130]: I found out I do not need it. The `cancelFunc` is called only on `close()`, that calls tracker's `close()` that cancels catchpoint generation as well. I updated the comment

File: ledger/ledger.go [hunk 4]
```diff
@@ -391,9 +399,19 @@ func (l *Ledger) notifyCommit(r basics.Round) basics.Round {
...
 }
 
+func (l *Ledger) waitAccountsWriting() {
+	l.trackers.waitAccountsWriting()
+}
+
+func (l *Ledger) scheduleCommit(rnd basics.Round) {
+	l.trackers.scheduleCommit(rnd)
+}
+
```
[the remark this answers is not in the document]
Comment (tsachiherman) [reply ledger/ledger.go:414]: I'm ok leaving it here for now - but it's not the right place for that. It looks to me as a "hint" that it doesn't really belong here. This part would "roll-forward" the database to be at the correct round ( i.e. x -> latest-320 ).

[the remark this answers is not in the document]
Comment (algorandskiy) [reply ledger/ledger.go:414]: I agree, it is part of account updates initialization, I did not figure out a better way to accessing upper level from an tracker.

**Review by tsachiherman — APPROVED** [review]
looks good.

**Pull request merged**